src/Elliptic/DiscontinuousGalerkin/SubdomainOperator/SubdomainOperator.hpp

```diff
@@ -232,31 +251,43 @@ struct SubdomainOperator
             const auto& map_keys, const auto... fields_and_fluxes) noexcept {
           constexpr bool is_overlap_v =
               std::decay_t<decltype(is_overlap)>::value;
-          const auto& boundary_conditions = local_domain.blocks()
-                                                .at(local_element_id.block_id())
-                                                .external_boundary_conditions();
-          ASSERT(boundary_conditions.contains(local_direction),
-                 "No boundary condition is available in block "
-                     << local_element_id.block_id() << " in direction "
-                     << local_direction
-                     << ". Make sure you are setting up boundary conditions "
-                        "when creating the domain.");
-          ASSERT(dynamic_cast<const BoundaryConditionsBase*>(
-                     boundary_conditions.at(local_direction).get()) != nullptr,
-                 "The boundary condition in block "
-                     << local_element_id.block_id() << " in direction "
-                     << local_direction
-                     << " has an unexpected type. Make sure it derives off the "
-                        "'boundary_conditions_base' class set in the system.");
+          // Get boundary conditions from domain, or use overridden boundary
+          // conditions
           const auto& boundary_condition =
-              dynamic_cast<const BoundaryConditionsBase&>(
-                  *boundary_conditions.at(local_direction));
+              [&local_domain, &local_element_id, &local_direction,
+               &override_boundary_conditions]() noexcept
+              -> const BoundaryConditionsBase& {
+            if (override_boundary_conditions.has_value()) {
+              return **override_boundary_conditions;
+            }
+            const auto& boundary_conditions =
+                local_domain.blocks()
+                    .at(local_element_id.block_id())
+                    .external_boundary_conditions();
+            ASSERT(boundary_conditions.contains(local_direction),
+                   "No boundary condition is available in block "
+                       << local_element_id.block_id() << " in direction "
+                       << local_direction
+                       << ". Make sure you are setting up boundary conditions "
+                          "when creating the domain.");
+            ASSERT(
+                dynamic_cast<const BoundaryConditionsBase*>(
+                    boundary_conditions.at(local_direction).get()) != nullptr,
+                "The boundary condition in block "
+                    << local_element_id.block_id() << " in direction "
+                    << local_direction
+                    << " has an unexpected type. Make sure it derives off the "
+                       "'boundary_conditions_base' class set in the system.");
+            return dynamic_cast<const BoundaryConditionsBase&>(
```

Comment:
it's a down-cast because the domain holds a generic `domain::BoundaryCondition`, and this casts it to a system-specific boundary-condition base class.